### DISCUSSION TITLE
Feature request: start phase-space from random particle

### DISCUSSION THREAD

**Discussion by skmartin** [nrc-cnrc/EGSnrc#696] [Ideas]
I was wondering if you would be able to help me with some information about how the phsp are read from in BEAMnrc. I want to use the Varian TrueBeam models (IAEAphsp) files as a source 21 and run the particles through collimator, etc.  I have been able to run them fine, but I want to cut back on time dedicated.  I wanted to know: When BEAM runs from source 21 and is reading the particles in the file, does it do so sequentially, no matter what the random seed is, or does it sample randomly from the phsp depending on seed.  My plan was to run on a cluster resource with 24 dedicated hour time windows and use the same phase space with different seeds to get more histories, however, if it is sampling sequentially, then I would not be sampling the IAEAphsp completely.

**Comment by rtownson** [comment]
Hi Stephen! If you have more questions along this line consider asking in the google+ community - there are a lot of members who have tackled similar problems.

https://plus.google.com/communities/106437507294474212197

BEAM does read sequentially from the phase-space, starting at the first particle in the file. When you parallelize using BEAM, it automatically splits up the phase-space so that each sub-field starts from a different (evenly divided) section of the phase-space. However, it does not depend on the random number seed. To reduce latent phase-space variance, you want to use as much of the phase-space as possible and parallelizing or reducing recycling can help with this. If that doesn't work for you, you could write a script to split up your phase-space beforehand into smaller phase-spaces. 

**I will suggest an enhancement** to the `EGSnrc` phase-space tools: add the option to start reading particles from a random location in the phase-space based on the seed. I have done this in a similar application and it can be quite effective.

**Comment by skmartin** [comment]
Thank you for the suggestions sir.
